akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala

```diff
@@ -613,6 +613,9 @@ final class GraphStageIsland(
 
   def conn(slot: Int): Connection = {
     maxConnections = math.max(slot, maxConnections)
+    if (maxConnections >= connections.length) {
+      connections = connections.padTo(connections.length * 2, null)
+    }
```

Comment:
The only motivation was so that it does not grow too many times. From all of the stream test cases 9 need to grow this array once in order to pass.